contracts/RewardManager.sol

```diff
@@ -319,13 +345,25 @@ contract RewardManager is Ownable, Versionable, Safe {
       msg.sender,
       rewardSafeOwner
     );
+
+    bool isAllowedGnosisExecution = _equalBytes(data, encodedTransactionData) &&
+      signatures[keccak256(contractSignature)] &&
+      isAllowedGnosisRecipient(to);
+
+    bytes4 result = isAllowedGnosisExecution ||
+      (to == address(this)) ||
```

Comment:
@alex-cardstack it looks like previously I missed out something. each or condition should check `_equalBytes(data,encodedTransactionData)`. Maybe just a `require(_equalBytes(data, encodedTransactionData)`. This is because we have to validate that `to` param is correct.